fmn/consumer/backends/sse.py

```diff
@@ -0,0 +1,180 @@
+        # Tack a human-readable delta on the end so users know that fmn is
+        # backlogged (if it is).
+        if msg['timestamp']:
+            date_time = msg['timestamp']
+            date_time = datetime.datetime.fromtimestamp(date_time)
+            date_time = date_time.isoformat()
+        else:
+            date_time = datetime.datetime.utcnow().isoformat()
+
+        icon = fedmsg.meta.msg2icon(msg, **config)
+        link = fedmsg.meta.msg2link(msg, **config)
+        secondary_icon = fedmsg.meta.msg2secondary_icon(msg, **config)
+        username = fedmsg.meta.msg2agent(msg, **config)
+        title = fedmsg.meta.msg2title(msg, **config)
+        subtitle = fedmsg.meta.msg2subtitle(msg, **config)
+
+    else:
+        # This handles messages that have already been 'conglomerated'.
+        title = u""
+        subtitle = msg['subtitle']
+        link = msg['link']
+
+    if recipient['shorten_links']:
+        link = shorten(link)
+    event_link = ''
+    if title and link:
+        event_link = '<a href="' + link + '">' + title + '</a>'
+    user_link = ''
+    if username:
+        user_link = '<a href="/'+str(username).replace("@", "")+'">'+username+'</a>'
+
+    markup = user_link + ' ' + event_link + ' ' + subtitle
+    output = {"dom_id": str(dom_id),
+              "date_time": str(date_time),
+              "icon": str(icon),
+              "link": str(link),
+              "markup": str(markup),
+              "secondary_icon": str(secondary_icon)
+              }
+
+    return json.dumps(output)
+
+
+class SSEBackend(BaseBackend):
+    __context_name__ = "sse"
+
+    def __init__(self, *args, **kwargs):
+        super(SSEBackend, self).__init__(*args, **kwargs)
+
+    def send_msg(self, msg="hello world", username="user.id.fedoraproject.org"):
+        username = str(username)
+        self.log.debug("SSE:    Sending msg to pika to queue " + username)
+        ms_24hr = 24 * 60 * 60 *1000
+        fq = FeedQueue(queue_name=username)
+        fq.push_message(msg)
+
+    def handle(self, session, recipient, msg, streamline=False):
+        self.log.debug("SSE:    handle")
+        short_message = _format_message(msg=msg, recipient=recipient, config=self.config)
+        user = recipient['user']
+        self.send_msg(short_message, user)
+
+    def handle_batch(self, session, recipient, queued_messages):
+        self.log.debug("SSE:    handle batch")
+        messages = [m.message for m in queued_messages]
+        # Squash some messages into one conglomerate message
+        # https://github.com/fedora-infra/datagrepper/issues/132
+        messages = fedmsg.meta.conglomerate(messages, **self.config)
+        for message in messages:
+            self.handle(session, recipient, message, streamline=True)
+
+    def handle_confirmation(self, session, confirmation):
+        self.log.debug("Handling confirmation sse %r" % confirmation)
+        pass
+
+
+class FeedQueue:
```

Comment:
I use this same class in fmn.sse, not sure If I should make this a separate file and link them somehow?